pkg/maps/ctmap/ctmap.go

```diff
@@ -840,3 +931,54 @@ func calculateInterval(prevInterval time.Duration, maxDeleteRatio float64) (inte
 
 	return
 }
+
+// CalculateCTMapPressure is a controller that calculates the BPF CT map
+// pressure and pubishes it as part of the BPF map pressure metric.
+func CalculateCTMapPressure(mgr *controller.Manager, allMaps ...*Map) {
+	ctx, cancel := context.WithCancelCause(context.Background())
+	mgr.UpdateController("ct-map-pressure", controller.ControllerParams{
+		Group: controller.Group{
+			Name: "ct-map-pressure",
+		},
+		DoFunc: func(context.Context) error {
+			var errs error
+			for _, m := range allMaps {
+				path, err := OpenCTMap(m)
+				if err != nil {
+					msg := "Skipping CT map pressure calculation"
+					scopedLog := log.WithError(err).WithField(logfields.Path, path)
+					if os.IsNotExist(err) {
+						scopedLog.Debug(msg)
+					} else {
+						scopedLog.Warn(msg)
+					}
+					continue
+				}
+				defer m.Close()
+
+				// The alternative is to use seq_ops from the kernel to dump
+				// the map. seq_file can be a buffer allocated from userspace
+				// which the kernel can write to. It's used to dump debug info
+				// mostly. But it requires the map to support BTF. No Cilium
```

Comment:
Do we have an issue for BTF support? Should we?

Reply:
Filed https://github.com/cilium/cilium/issues/29216. Empty now while I gather info, feel free to add if you have context already.

Reply:
I've removed the comment because it's not really that relevant to the reader anymore.

Initially, I had it from a discussion with Lorenz where he proposed `seq_file` as an alternative to batch ops.